container/tabs.go

```diff
@@ -0,0 +1,675 @@
+	r.close.Resize(fyne.NewSize(theme.IconInlineSize(), theme.IconInlineSize()))
+}
+
+func (r *tabButtonRenderer) MinSize() fyne.Size {
+	var contentWidth, contentHeight float32
+	textSize := r.label.MinSize()
+	if r.button.iconPosition == buttonIconTop {
+		contentWidth = fyne.Max(textSize.Width, r.iconSize())
+		if r.icon != nil {
+			contentHeight += r.iconSize()
+		}
+		if r.label.Text != "" {
+			if r.icon != nil {
+				contentHeight += theme.Padding()
+			}
+			contentHeight += textSize.Height
+		}
+	} else {
+		contentHeight = fyne.Max(textSize.Height, r.iconSize())
+		if r.icon != nil {
+			contentWidth += r.iconSize()
+		}
+		if r.label.Text != "" {
+			if r.icon != nil {
+				contentWidth += theme.Padding()
+			}
+			contentWidth += textSize.Width
+		}
+	}
+	if r.button.onClosed != nil {
+		contentWidth += theme.IconInlineSize() + theme.Padding()
+		contentHeight = fyne.Max(contentHeight, theme.IconInlineSize())
+	}
+	return fyne.NewSize(contentWidth, contentHeight).Add(r.padding())
+}
+
+func (r *tabButtonRenderer) Objects() []fyne.CanvasObject {
+	return r.objects
+}
+
+func (r *tabButtonRenderer) Refresh() {
+	if r.button.hovered {
+		r.background.FillColor = theme.HoverColor()
+		r.background.Show()
+	} else {
+		r.background.Hide()
+	}
+	r.background.Refresh()
+
+	r.label.Text = r.button.text
+	r.label.Alignment = r.button.textAlignment
+	if r.button.importance == widget.HighImportance {
+		r.label.Color = theme.PrimaryColor()
+	} else {
+		r.label.Color = theme.TextColor()
+	}
+	r.label.TextSize = theme.TextSize()
+	if r.button.text == "" {
+		r.label.Hide()
+	} else {
+		r.label.Show()
+	}
+
+	if r.icon != nil && r.icon.Resource != nil {
+		switch res := r.icon.Resource.(type) {
+		case *theme.ThemedResource:
+			if r.button.importance == widget.HighImportance {
+				r.icon.Resource = theme.NewPrimaryThemedResource(res)
+				r.icon.Refresh()
+			}
+		case *theme.PrimaryThemedResource:
+			if r.button.importance != widget.HighImportance {
+				r.icon.Resource = res.Original()
+				r.icon.Refresh()
+			}
+		}
+	}
+
+	if d := fyne.CurrentDevice(); r.button.onClosed != nil && (d.IsMobile() || r.button.hovered || r.close.hovered) {
+		r.close.Show()
+	} else {
+		r.close.Hide()
+	}
+	r.close.Refresh()
+
+	canvas.Refresh(r.button)
+}
+
+func (r *tabButtonRenderer) iconSize() float32 {
+	switch r.button.iconPosition {
+	case buttonIconTop:
+		return 2 * theme.IconInlineSize()
+	default:
+		return theme.IconInlineSize()
+	}
+}
+
+func (r *tabButtonRenderer) padding() fyne.Size {
+	if r.label.Text != "" && r.button.iconPosition == buttonIconInline {
+		return fyne.NewSize(theme.Padding()*4, theme.Padding()*2)
+	}
+	return fyne.NewSize(theme.Padding()*2, theme.Padding()*2)
+}
+
+var _ fyne.Widget = (*tabCloseButton)(nil)
+var _ fyne.Tappable = (*tabCloseButton)(nil)
+var _ desktop.Hoverable = (*tabCloseButton)(nil)
+
+type tabCloseButton struct {
```

Comment:
Did you consider a button with an icon and Importance = LowImportance?

Reply:
When `tabButton` is hovered the `tabCloseButton` becomes visible, but even after `tabButton` receives `MouseOut` the `tabCloseButton` must remain visible if it is hovered. `widget.Button.hovered` is not accessible from the `container` package.

Reply:
(container/tabs.go#540)

Reply:
Thanks